cli/core/pkg/command/context.go

```diff
@@ -64,7 +65,7 @@ func init() {
 
 	initCreateCtxCmd()
 
-	listCtxCmd.Flags().StringVarP(&ctxType, "type", "t", "", "context type (k8s|tmc)")
+	listCtxCmd.Flags().StringVarP(&target, "target", "t", "", "list only contexts associated with the specified target (kubernetes[k8s]|mission-control[tmc])")
```

Comment:
Need to support `k8s/tmc` inputs. Looks like it doesn't support it right now. Also need a validation if user enters invalid target, it should throw and error. Currently, it returns empty list.

```
$ tanzu context list -t k8s
```
```
$ tanzu context list -t kubernetes
Target:  kubernetes
  NAME                 ISACTIVE  ENDPOINT  KUBECONFIGPATH             KUBECONTEXT        
  kind-test-cluster    false               /Users/anujc/.kube/config  kind-test-cluster  
  kind-test-cluster-2  true                /Users/anujc/.kube/config  kind-test-cluster  
  ```
  ```
  $ tanzu context list -t random
```